lib/ui/text.dart

```diff
@@ -658,6 +663,7 @@ class ParagraphStyle {
     FontWeight fontWeight,
     FontStyle fontStyle,
     StrutStyle strutStyle,
+    TextWidthType widthType,
```

Comment:
So since dart:ui is a more base level library, we may want to consider handline the switching between full and tight completely in the framework as an abstraction over dart:ui. At the dart:ui, I think it makes more sense for both widths to be available in the getter for the width. Either make the width getter property take the enum or make a separate getter.

The idea of innately possessing a tight or full width is more of a framework-level issue, and dart:ui (and below) shouldn't really have the paragraph preemptively pick its width as it would now with a setting in ParagraphStyle.

Reply:
I think it makes sense for the setting to be in ParagraphStyle, but have the framework consult it or otherwise extract the widthType from the ParagraphStyle to choose with dart:ui method to call.

This would mean it isnt necessary to pass the widthType all the way down into Libtxt in the initial passing of ParagraphStyle into native.